dotCMS/src/main/java/com/dotcms/security/secret/ServiceIntegrationAPIImpl.java

```diff
@@ -0,0 +1,554 @@
+        final File basePath = new File(ymlFilesPath);
+        if (!basePath.exists()) {
+            basePath.mkdir();
+        }
+        Logger.debug(ServiceIntegrationAPIImpl.class,
+                () -> " ymlFiles are set under:  " + ymlFilesPath);
+        final Set<String> files = listFiles(ymlFilesPath);
+        if (!UtilMethods.isSet(files)) {
+            // No files were found a default empty descriptor will be created.
+            // ymlMapper.writeValue(new File(basePath, "sample-service-descriptor.yml"), emptyDescriptor());
+            // return listFiles(ymlFilesPath);
+           return Collections.emptySet();
+        } else {
+            return files;
+        }
+    }
+
+    private Set<String> listFiles(final String dir) throws IOException {
+        final Set<String> fileList = new HashSet<>();
+        try (final DirectoryStream<Path> stream = Files.newDirectoryStream(Paths.get(dir))) {
+            for (final Path path : stream) {
+                if (!Files.isDirectory(path)) {
+                    final String fileName = path.getFileName().toString();
+                    if (fileName.endsWith("yaml") || fileName.endsWith("yml")) {
+                        fileList.add(path.toString());
+                    }
+                }
+            }
+        }
+        return fileList;
+    }
+
+    private List<ServiceDescriptorMeta> loadServiceDescriptors()
+            throws IOException, DotDataException {
+        final Set<String> loadedServiceKeys = new HashSet<>();
+        final ImmutableList.Builder<ServiceDescriptorMeta> builder = new ImmutableList.Builder<>();
+        final Set<String> fileNames = listAvailableYamlFiles();
+        for (final String fileName : fileNames) {
+            try {
+                final File file = new File(fileName);
+                final ServiceDescriptor serviceDescriptor = ymlMapper
+                        .readValue(file, ServiceDescriptor.class);
+                if (validateServiceDescriptor(serviceDescriptor)) {
+                    if (loadedServiceKeys.contains(serviceDescriptor.getKey())) {
+                        throw new DotDataException(
+                                String.format(
+                                        "There's a service already registered under key `%s`.",
+                                        serviceDescriptor.getKey())
+                                );
+                    }
+                    builder.add(new ServiceDescriptorMeta(serviceDescriptor, file.getName()));
+                    loadedServiceKeys.add(serviceDescriptor.getKey());
+                }
+            } catch (IOException e) {
+                Logger.error(ServiceIntegrationAPIImpl.class,
+                        String.format("Error reading yml file `%s`.", fileName), e);
+            }
+        }
+
+        return builder.build();
+    }
+
+   private boolean validateServiceDescriptor(final ServiceDescriptor serviceDescriptor)
```

Comment:
I think the return is not necessary since everything is handle thought exceptions